### PR TITLE
Chore/type errors in events feature

### PR DESCRIPTION
**Pull request by dakeyras7**
This PR does a couple of things:

* Introduce the correct typing package for styled-components (type errors 270 -> 415)
* Fix all type issues related to the mapping event feature (type errors 415 -> 370)
* Fix all obvious typing errors related to browser events and missing/wrong props in styled components that were masked before (type errors 370 -> 313)

### CODE REVIEW

File: src/components/MappingEvents/MappingEventShareBar.js [hunk 1]
```diff
@@ -12,10 +29,11 @@ const MappingEventShareBar = ({ className, mappingEvent, buttonCaption, baseUrl,
   const description = mappingEvent.description || mappingEvent.name;
 
   const mailSubject = sharedObjectTitle;
+  const productNameString = productName ? ` on ${productName}` : '';
   // translator: Email text used when sharing a mapping event via email.
   let mailBody = t`Help us out and join the ${
     mappingEvent.name
-  } mapping event on ${productName}. You can find more info here: ${url}`;
+  } mapping event${productNameString}. You can find more info here: ${url}`;
```
Comment (mutaphysis) [inline src/components/MappingEvents/MappingEventShareBar.js]: 0: Readd space?

File: src/components/MappingEvents/Statistics.js [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
-  mappedPlacesCount: number;
-  invitedParticipantCount: number;
-  startDate: Date;
-  endDate: ?Date;
-  className: string;
-}
+type Props = {
+  mappedPlacesCount: number,
+  invitedParticipantCount: number,
+  startDate: Date,
```
Comment (mutaphysis) [inline src/components/MappingEvents/Statistics.js]: -1: startDate is optional in the mapping event planner

File: src/components/MappingEvents/MappingEventShareBar.js (hunk 1 above)
Comment (dakeyras7) [reply src/components/MappingEvents/MappingEventShareBar.js]: The space is now part of the `productNameString` so we don't have an extraneous space when `productName` does not exist.

File: src/components/MappingEvents/Statistics.js (hunk 2 above)
Comment (dakeyras7) [reply src/components/MappingEvents/Statistics.js]: Done

File: src/components/MappingEvents/MappingEventShareBar.js (hunk 1 above)
Comment (mutaphysis) [reply src/components/MappingEvents/MappingEventShareBar.js]: Cool!

**Pull request merged**